back-end/src/sciler/handler/handler.go

```diff
@@ -166,68 +171,181 @@ func (handler *Handler) onInstructionMsg(raw Message) {
+	}
+	handler.sendEventStatus()
+}
+
+// onResetAll is the function to process the instruction `reset all`
+// reset all is instructed when the reset button is clicked in the front-end
+func (handler *Handler) onResetAll(deviceID string) {
+	handler.sendInstruction("client-computers", []map[string]string{{
+		"instruction":   "reset",
+		"instructed_by": deviceID,
+	}})
+	handler.sendInstruction("front-end", []map[string]string{{
+		"instruction":   "reset",
+		"instructed_by": deviceID,
+	}})
+	handler.Config = config.ReadFile(handler.ConfigFile)
+	handler.SendSetup()
+}
+
+// onTestAll is the function to process the instruction `test all`
+// test all is instructed when the test button is clicked in the front-end
+func (handler *Handler) onTestAll(instructor string) {
+	handler.sendInstruction("client-computers", []map[string]string{{
+		"instruction":   "test",
+		"instructed_by": instructor,
+	}})
+}
+
+// onTestDevice is the function to process the instruction `test device`
+// test device is instructed when the test button of a device is clicked in the front-end
+func (handler *Handler) onTestDevice(deviceID string, instructor string) {
+	handler.sendInstruction(deviceID, []map[string]string{{
+		"instruction":   "test",
+		"instructed_by": instructor,
+	}})
+}
+
+// onResetAll is the function to process the instruction `finish rule`
+// finish rule is instructed then the "voer uit" button of a rule is clicked in the front-end
+func (handler *Handler) onFinishRule(ruleID string) {
+	rule, ok := handler.Config.RuleMap[ruleID]
+	if !ok {
+		logger.Errorf("could not find rule with id %s in map", ruleID)
+	} else {
+		rule.Execute(handler)
+	}
+	handler.sendEventStatus()
+}
+
+// onHint is the function to process the instruction `hint`
+// hint in instructed when the front-end submits a hint
+func (handler *Handler) onHint(hint string, instructor string) {
+	handler.sendInstruction("hint", []map[string]string{{
+		"instruction":   "hint",
+		"value":         hint,
+		"instructed_by": instructor,
+	}})
+}
+
+// onCheckConfig is the function to process the instruction `check config`
+// check the config and sends a message containing all errors it could find
+func (handler *Handler) onCheckConfig(configToRead interface{}) {
+	message := Message{
+		DeviceID: "back-end",
+		TimeSent: time.Now().Format("02-01-2006 15:04:05"),
+		Type:     "config",
+		Contents: map[string][]string{"errors": handler.checkConfig(configToRead)},
+	}
+	jsonMessage, _ := json.Marshal(&message)
+	handler.Communicator.Publish("front-end", string(jsonMessage), 3)
+}
+
+// processConfig reads the config in.
+// If action is "use" then the message must tell the config a new config is now used and put it to use
+func (handler *Handler) onUseConfig(configToRead interface{}, fileName string) {
+	// check if the config can be used, if so use and send message
+	if len(handler.checkConfig(configToRead)) == 0 {
+		handler.useConfig(configToRead, fileName)
```

Comment:
now you're first sending set up and then the actual new config message, so change the order here. 

first write file and set config in handler, then send new config then send status ( i know this wasn't the case earlier but this seems better)

Reply:
fixed